include/envoy/config/BUILD

```diff
@@ -44,7 +44,7 @@ envoy_cc_library(
     hdrs = ["subscription_factory.h"],
     deps = [
         ":subscription_interface",
-        "@envoy_api//envoy/api/v2/core:pkg_cc_proto",
+        "@envoy_api_shadow//envoy/api/v2/core:pkg_cc_proto",
```

Comment:
can we switch the bind, (i.e. use `@envoy_api` for shadow)? I think that makes less breaking for BUILD files under `include/` and `source/`, and less breaking for consuming project, as @mattklein123 pointed out.

Otherwise this will easily cause ODR violation when some consuming project trying to link `@envoy_api//envoy/api/v2/core:pkg_cc_proto`.

Reply:
I have done this, but it comes with tradeoffs. While most of the source now deals with `@envoy_api`, we have to switch all developer tools to an alias for the real artifacts, `@envoy_api_published` (lest we have some weird dependence of generating tools on generated artifacts). Then we need to symlink across Bazel rules. It works, PTAL.

Reply:
Yeah it comes with tradeoffs but I prefer this way to be less breaking for consuming projects.